project1/server/server_tasks/agent_task.cpp

```diff
@@ -0,0 +1,35 @@
+#include "agent_task.h"
+
+namespace server_tasks {
+
+    void AgentTask::SetClientFD(int fd) {
+        client_fd_ = fd;
+    }
+
+    void AgentTask::SetActiveCommands(std::shared_ptr<CommandIDs>cmds) {
+        active_commands_ = cmds;
+    }
+
+    void AgentTask::SetFileAccessManagers(std::shared_ptr<server::file_handler::FileAccessManager> read_mgr,
+                                          std::shared_ptr<server::file_handler::FileAccessManager> write_mgr) {
+        read_manager_ = read_mgr;
+        write_manager_ = write_mgr;
+    }
+
+    thread_pool::Task::Status AgentTask::SetUp() {
+        // give the agent a unique file handler with the shared access managers
+        auto fh = std::make_unique<server::file_handler::ThreadSafeFileHandler>(read_manager_,write_manager_);
+        agent_ = new server::Agent(client_fd_, std::move(fh),active_commands_);
```

Comment:
Don't use `new`, it will lead to memory leaks as it does in this case since you never call `delete`. Use a `unique_ptr` instead.

Reply:
So, I tried to do this initially. I kept getting a weird error with type conversion that I'm not sure how to deal with. 

This line: `agent_ = std::unique_ptr<server::Agent>(client_fd_, std::move(fh), active_commands_);`

produces this error `Assigning to 'server::Agent *' from incompatible type 'std::unique_ptr<server::Agent>'`